docs/false_positives.md

```diff
@@ -0,0 +1,243 @@
+Table of Contents
+=================
+* [False positives](#false-positives)
+  * [What target to analyze?](#what-target-to-analyze)
+  * [Infeasible path](#infeasible-path)
+    * [Correlated conditions](#correlated-conditions)
+    * [Partial functions](#partial-functions)
+    * [Prefer standard functions](#prefer-standard-functions)
+    * [Do not turn off code checks](#do-not-turn-off-code-checks)
+    * [Suppress specific dead store warnings](#suppress-specific-dead-store-warnings)
+    * [Alternative implementations](#alternative-implementations)
+  * [Syntax based checks](#syntax-based-checks)
+  * [Suppress results](#suppress-results)
+    * [3rd party code](#third-party-code)
+    * [Authored code](#authored-code)
+    * [Unauthored code](#unauthored-code)
+
+## <a name="false-positives"></a> False positives
+
+CodeChecker is running static analysis tools on the code. Unfortunately, it is
+not possible to create perfect tools. They might report correct code as
+incorrect. These findings are called false positives. This document intended
+to help how to deal with them. As a rule of thumb, whenever you encounter a
+false positive finding using suppression should be only the last resort. Why?
+Read on.
+
+Having a false positive indicates that the analyzer do not understand some
+properties of the code. Suppressing a result will not help its understanding.
+Making the code more obvious for the tool, however, makes the analysis more
+precise. As a bonus, such code sometimes also more readable for developers.
+
+This guide introduce tips and tricks how to make the code easier to analyze.
+
+### <a name="what-target-to-analyze"></a> What target to analyze?
+
+Usually, a project has multiple build targets for different purposes. There
+might be multiple targets for multiple architectures, releases, debugging.
+
+We advise to use the debug target to analyze the code. The reason is, debug
+targets usually contains assertions. These assertions convey useful information
```

Comment:
For the sake of understanding, perhaps we should explicitly mention in a subsequence that while `DEBUG` contains assertions, `RELEASE` usually **does not**, so adding assertions don't slow your code down *in production*.